File: src/lib/format.ts
// Formatage des prix et des dates en français (Côte d'Ivoire).

/** Espace insécable : garde le nombre et son unité (FCFA) sur la même ligne. */
const NBSP = ' '

/** Nombre groupé à la française : « 1 500 » (séparateur = espace fine insécable). */
export function formatPrice(value: number): string {
  return new Intl.NumberFormat('fr-FR').format(value)
}

/** Prix complet à la française : « 1 500 FCFA » (espace insécable avant l'unité). */
export function formatFCFA(value: number): string {
  return `${formatPrice(value)}${NBSP}FCFA`
}

export function priceLabel(value: number, negotiable = false): string {
  if (value === 0) return 'Gratuit'
  return `${formatFCFA(value)}${negotiable ? `${NBSP}— à débattre` : ''}`
}

export function timeAgo(timestamp: number): string {
  const diff = Date.now() - timestamp
  const min = Math.floor(diff / 60000)
  if (min < 1) return "À l'instant"
  if (min < 60) return `il y a ${min} min`
  const h = Math.floor(min / 60)
  if (h < 24) return `il y a ${h} h`
  const d = Math.floor(h / 24)
  if (d < 7) return `il y a ${d} j`
  const w = Math.floor(d / 7)
  if (w < 5) return `il y a ${w} sem.`
  const mo = Math.floor(d / 30)
  if (mo < 12) return `il y a ${mo} mois`
  return `il y a ${Math.floor(d / 365)} an(s)`
}

File: flutter_app/lib/format.dart
/// Formatage à la française — les mêmes règles que le site.
///
/// Prix en FCFA avec l'espace insécable pour séparer les milliers et avant
/// « FCFA » (1 500 FCFA, jamais 1500FCFA ni 1 500 FCFA cassé en fin de ligne).
library;

const String _nbsp = ' '; // espace insécable

/// 1500 -> « 1 500 FCFA ». Le séparateur de milliers est insécable.
String formatFCFA(num montant) {
  final entier = montant.round().abs().toString();
  final tampon = StringBuffer();
  for (int i = 0; i < entier.length; i++) {
    if (i > 0 && (entier.length - i) % 3 == 0) tampon.write(_nbsp);
    tampon.write(entier[i]);
  }
  final signe = montant < 0 ? '-' : '';
  return '$signe$tampon${_nbsp}FCFA';
}

/// Heure locale courte « 14:05 » à partir d'un timestamp en millisecondes.
String heureCourte(int ms) {
  final d = DateTime.fromMillisecondsSinceEpoch(ms).toLocal();
  final h = d.hour.toString().padLeft(2, '0');
  final m = d.minute.toString().padLeft(2, '0');
  return '$h:$m';
}

/// Temps écoulé depuis un timestamp en millisecondes -> « il y a 3 h », « hier »…
String tempsEcoule(int timestampMs) {
  final maintenant = DateTime.now().millisecondsSinceEpoch;
  final secondes = ((maintenant - timestampMs) / 1000).round();
  if (secondes < 60) return "à l'instant";
  final minutes = (secondes / 60).round();
  if (minutes < 60) return 'il y a $minutes${_nbsp}min';
  final heures = (minutes / 60).round();
  if (heures < 24) return 'il y a $heures${_nbsp}h';
  final jours = (heures / 24).round();
  if (jours == 1) return 'hier';
  if (jours < 7) return 'il y a $jours${_nbsp}jours';
  final semaines = (jours / 7).round();
  if (semaines < 5) return 'il y a $semaines${_nbsp}sem.';
  final mois = (jours / 30).round();
  if (mois < 12) return 'il y a $mois${_nbsp}mois';
  return 'il y a ${(jours / 365).round()}${_nbsp}an(s)';
}
